### ISSUE TITLE
The slave catchup failure

### ISSUE THREAD

**Issue by ghost** [y-trudeau/resource-agents-prm#1]
sorry to use this way to ask:-)

I was exploring your enhancements to mysql resource agent. I'm little bit puzzled by loops in unset_master function. 
https://github.com/y-trudeau/resource-agents-prm/blob/master/heartbeat/mysql#L612

In situation when one of the slaves is too far from master and catching up could take a few minutes, could this resource action fail due to timeout and consequently cause the failure of promotion step on new master node?

**Comment by y-trudeau** [comment]
Yes, it could fail.  You just can't failover to a slave that is lagging 
behind by too much, data consistency is at risk.

Regards,

Yves

**Comment by ghost** [comment]
Thanks for response:)

Ok, that makes total sense.

To put it differently: could long running "mysql_notify -> unset_master" block promotion of the new master? Because I was living at the impression that the chain

pre-demote
demote
post-demote
pre-promote
promote
post-promote

is running sequentially across all nodes of the cluster. So if one steps stall, all of in the sequence stall too.

**Comment by y-trudeau** [comment]
Yes, it could block the promotion but in that case, keep in mind that the master_score will be lower for that slave if it lags behind so it is less likely to be the next master.
